### ISSUE TITLE
[DataGrid] Support multiple selection modes

### ISSUE THREAD

**Issue by woodlandhunter** [objorke/PropertyTools#153]
Would you consider a change to the DataGrid to allow different selection modes?

At the very least `Extended`/`Single` could be supported [like the regular datagrid](https://msdn.microsoft.com/en-us/library/system.windows.controls.datagridselectionmode(v=vs.110).aspx).

But an approach with added modes could also be taken:

`Extended`
`Single`
`SameRow` (only cells on the same row can be multi-selected)
`SameColumn` (only cells on the same column can be multi-selected)
`SameRowOrColumn` (only one direction at a time)

What do you think?

**Comment by objorke** [comment]
Yes, this is something I have also been wanting. A multi-select (extended) selection mode would be great. 
I have not though of same row/column functionality - do you need this?

It would be great if it could work like in Excel:
![image](https://cloud.githubusercontent.com/assets/439716/22100472/0f3c9fdc-de30-11e6-9f12-18b09dbada74.png)

**Comment by woodlandhunter** [comment]
Based on some rough testing I think implementing any of the contiguous selection modes should be fairly simple.  However I don't need same row/column functionality currently (I only currently need single cell).

Multiple selections seems like a bigger refactor.
